adding legacy template api and re-numbering the index API overall (#10330)
opensearch-project/documentation-website: _api-reference/index-apis/post-template-legacy.md

---
layout: default
title: Post template (deprecated)
parent: Index APIs
nav_order: 107
---

> Post template API is deprecated. Use the new [Create or update index template]({{site.url}}{{site.baseurl}}/api-reference/index-apis/create-index-template/) API.

Reviewer: Move this to after the heading.

> Post template API is deprecated. Use the new [Create or update index template]({{site.url}}{{site.baseurl}}/api-reference/index-apis/create-index-template/) API.

Reviewer: ```suggestion
Post template API is deprecated. Use the new [Create or Update Index Template]({{site.url}}{{site.baseurl}}/api-reference/index-apis/create-index-template/) API.
```

> Post Template API is deprecated. Use the new [Create or Update Index Template]({{site.url}}{{site.baseurl}}/api-reference/index-apis/create-index-template/) API.

Reviewer 2: ```suggestion
The Post Template API has been deprecated. Use the new [Create or Update Index Template]({{site.url}}{{site.baseurl}}/api-reference/index-apis/create-index-template/) API.
```

{: .warning}

The post template API operation is used to create or update an index template. Templates define settings, mappings, and aliases applied automatically when a matching index is created.

## Endpoints

```json
POST /_template/<template-name>
```

## Path parameters

All path parameters are required.

| Parameter       | Type   | Description                                     |
| :-------------- | :----- | :---------------------------------------------- |
| `template-name` | String | The name of the index template to create or update. |

## Query parameters

All query parameters are optional.

| Parameter        | Type    | Description                                                                                                       |
| :--------------- | :------ | :---------------------------------------------------------------------------------------------------------------- |
| `order`          | Integer | The order in which to apply the template if multiple templates match. Higher values are applied last. Default is `0`. |

> | `create`         | Boolean | If true, the operation fails if a template with the same name already exists. Default is `false`.             |

Reviewer 2: ```suggestion
| `create`         | Boolean | If `true`, the operation fails if a template with the same name already exists. Default is `false`.             |
```

> | `create`         | Boolean | If true, the operation will fail if a template with the same name already exists. Default is `false`.             |

Reviewer: ```suggestion
| `create`         | Boolean | If true, the operation fails if a template with the same name already exists. Default is `false`.             |
```

> | `cluster_manager_timeout` | Time    | The time to wait for connection to the cluster manager node. Default is `30s`.                                        |

Reviewer 2: ```suggestion
| `cluster_manager_timeout` | Time    | Specifies how long to wait for a connection to the cluster manager node. Default is `30s`.                                        |
```

## Request body

> The request body must define one or more of the following components:

Reviewer 2: ```suggestion
The request body must define one or more of the following components.
```

| Field            | Type   | Description                                                          |
| :--------------- | :----- | :------------------------------------------------------------------- |
| `index_patterns` | Array  | The list of index name patterns to which the template applies. Required. |
| `settings`       | Object | The index settings to apply to matching indexes.                         |
| `mappings`       | Object | The mappings for fields in the index.                                    |
| `aliases`        | Object | The aliases to assign to matching indexes.                               |

> | `version`        | Integer          | The optional version number to identify the template.                   |

Reviewer 2: ```suggestion
| `version`        | Integer          | The optional version number used to identify the template.                   |
```
